src/test/java/com/checkmarx/jira/JiraTestUtils.java

```diff
@@ -0,0 +1,337 @@
+import java.util.regex.Pattern;
+
+
+@TestComponent
+public class JiraTestUtils implements IJiraTestUtils {
+    private static final Logger log = LoggerFactory.getLogger(JiraTestUtils.class);
+
+    private static final String JIRA_DESCRIPTION_FINDING_LINE = "[Line #";
+
+    private JiraRestClient client;
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @Autowired
+    private JiraProperties jiraProperties;
+
+    @PostConstruct
+    public void initClient() {
+        if (jiraProperties != null && !ScanUtils.empty(jiraProperties.getUrl())) {
+            CustomAsynchronousJiraRestClientFactory factory = new CustomAsynchronousJiraRestClientFactory();
+            URI jiraURI = null;
+            try {
+                jiraURI = new URI(jiraProperties.getUrl());
+            } catch (URISyntaxException e) {
+                //log.error("Error constructing URI for JIRA", e);
+            }
+            this.client = factory.createWithBasicHttpAuthenticationCustom(jiraURI, jiraProperties.getUsername(), jiraProperties.getToken(), jiraProperties.getHttpTimeout());
+
+        }
+    }
+
+    @Override
+    public void deleteIssue(String issueKey) {
+        client.getIssueClient().deleteIssue(issueKey, true).claim();
+    }
+
+    private SearchResult search(String jql) {
+        return  client.getSearchClient().searchJql(jql).claim();
+    }
+
+    @Override
+    public void cleanProject(String projectKey) {
+        SearchResult searchResult = search(String.format("project = \"%s\"", projectKey));
+        for (Issue issue: searchResult.getIssues()) {
+            deleteIssue(issue.getKey());
+        }
+    }
+
+    @Override
+    public int getNumberOfIssuesInProject(String projectKey) {
+        SearchResult result = search(String.format("project = \"%s\"", projectKey));
+        return result.getTotal();
+    }
+
+    @Override
+    public Map<Filter.Severity, Integer> getIssuesPerSeverity(String projectKey) {
+        Map<Filter.Severity, Integer> result= new HashMap<>();
+        SearchResult searchResults = search(String.format("project = \"%s\"", projectKey));
+        for (Issue issue: searchResults.getIssues()) {
+            String severity = getIssueSeverity(issue.getDescription()).toUpperCase();
+            Filter.Severity filterSeverity = Filter.Severity.valueOf(severity.toUpperCase());
+            if (severity != null) {
+                if (result.containsKey(filterSeverity)) {
+                    result.put(filterSeverity,result.get(filterSeverity) + 1 );
+                } else {
+                    result.put(filterSeverity, 1);
+                }
+            }
+        }
+        return result;
+    }
+
+    private String getIssueSeverity(String issueDescription) {
+        return getIssueBodyPart(issueDescription,"Severity:");
+    }
+
+    private String getIssueBodyPart(String issueDescription, String field) {
+        String[] lines = issueDescription.split(System.lineSeparator());
```

Comment:
Can the actual separator be different from System.lineSeparator() ?

Reply:
I don't know for sure.
However, I do know that in windows it is CRLF, and in other UNIX-like it is not, so I used this.